protocol/contracts/ecosystem/ShipmentPlanner.sol

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+/**
+ * @notice Constraints of how many Beans to send to a given route at the current time.
+ * @param points Weight of this shipment route relative to all routes. Expects precision of 1e18.
+ * @param cap Maximum Beans that can be received by this stream at this time.
+ */
+struct ShipmentPlan {
+    uint256 points;
+    uint256 cap;
+}
+
+interface IBeanstalk {
+    function isFertilizing() external view returns (bool);
+    function totalUnfertilizedBeans() external view returns (uint256);
+
+    function isHarvesting(uint256 fieldId) external view returns (bool);
+    function totalUnharvestable(uint256 fieldId) external view returns (uint256);
+}
+
+uint256  constant BARN_POINTS = 333_333_333_333_333_333;
+uint256  constant FIELD_POINTS = 333_333_333_333_333_333;
+uint256  constant SILO_POINTS = 333_333_333_333_333_333;
+
+/**
+ * @title ShipmentPlanner
+ * @notice Contains getters for retrieving ShipmentPlans for various Beanstalk components.
+ * @dev Lives as a standalone immutable contract. Updating shipment plans requires deploying
+ *      a new instance and updating the ShipmentRoute planContract addresses help in AppStorage.
+ */
+contract ShipmentPlanner {
+    address constant BEANSTALK = 0xC1E088fC1323b20BCBee9bd1B9fC9546db5624C5;
```

Comment:
can we make this have a constructor with an immutable variable BEANSTALK? This would allow beanstalk forks to be spun up without changing `shipmentPlanners` (would be nice for an potential L2 migration)

Reply:
updated here. do not need to store the address anywhere
https://github.com/BeanstalkFarms/Beanstalk/pull/860/commits/d65b6bacc7e6d96105a48c759af2f5222b0ccb26#diff-cf3f4898f41eee71e7b37cb1f5e642c532b4efd2ce1ee23de550f167e61ea6b5R38-R40